thunderfish/consistentfishes.py

```diff
@@ -0,0 +1,148 @@
+
+    :param fishlists:       (4-D array) List of fishlists with harmonics and each frequency and power.
+                            fishlists[fishlist][fish][harmonic][frequency, power]
+    :return fundamentals:   (2-D array) list of lists containing the fundamentals of a fishlist.
+                            fundamentals = [ [f1, f1, ..., f1, f1], [f2, f2, ..., f2, f2], ..., [fn, fn, ..., fn, fn] ]
+    """
+    fundamentals = [[] for i in np.arange(len(fishlists))]
+    for fishlist in np.arange(len(fishlists)):
+        for fish in np.arange(len(fishlists[fishlist])):
+            fundamentals[fishlist].append(fishlists[fishlist][fish][0][0])
+
+    return fundamentals
+
+def find_consistency(fundamentals):
+    """
+    Compares several lists to find values that are, with a certain threshhold, available in every list.
+
+    This function gets a list of lists conatining values as input. For every value in the first list this function is
+    looking in the other lists if there is a entry with the same value +- threshold. The vector consistancy_help
+    has the length of the first list of fundamentals and it's entries are all ones. When a value from the first list is
+    also available in another list this function adds 1 to that particular entry in the consostany_help vector. After
+    comparina all values from the first list with all values of the other lists the consostancy_help vector got values
+    between 1 and len(fundamentals). The indices where this vector is equal to len(fundamentals) are also the indices of
+    fishes in the first fishlist that are available in every fishlist.
+
+
+    :param fundamentals:    (2-D array) list of lists containing the fundamentals of a fishlist.
+                            fundamentals = [ [f1, f1, ..., f1, f1], [f2, f2, ..., f2, f2], ..., [fn, fn, ..., fn, fn] ]
+    :return consistent_fundamentals: (1-D array) List containing all values that are available in all given lists.
+    :return index:          (1-D array) Indices of the values that are in every list relating to the fist list in fishlists.
+    """
+    consistancy_help = [1 for i in np.arange(len(fundamentals[0]))]
+    index = []
+    consistent_fundamentals = []
+
+    for enu, fundamental in enumerate(fundamentals[0]):
+        for list in np.arange(len(fundamentals)-1)+1:
+            freq_diff = [fundamentals[list][i] - fundamental for i in np.arange(len(fundamentals[list]))]
+            for i in np.arange(len(freq_diff)):
+                if freq_diff[i] >= -1.0 and freq_diff[i] <= 1:
+                    consistancy_help[enu] += 1
+                    break
+
+    for idx in np.arange(len(consistancy_help)):
+        if consistancy_help[idx] == len(fundamentals):
+            index.append(idx)
+            consistent_fundamentals.append(fundamentals[0][idx])
+
+    return consistent_fundamentals, index
+
+def consistent_fishlist(index, fishlists):
```

Comment:
-- I do not intend to be rude! --
Do we really need such an extensive docu for 3 lines of code? By reading it, I actually got more confused than when actually reading the code.  Correct me if I'm wrong!